astropy/stats/sigma_clipping.py

```diff
@@ -879,6 +881,287 @@ def sigma_clip(
+        until the last iteration clips nothing). If convergence is
+        achieved prior to ``maxiters`` iterations, the clipping
+        iterations will stop. The default is 5.
+
+    cenfunc : {'median', 'mean'} or callable, optional
+        The statistic or callable function/object used to compute
+        the center value for the clipping. If using a callable
+        function/object and the ``axis`` keyword is used, then it must
+        be able to ignore NaNs (e.g., `numpy.nanmean`) and it must have
+        an ``axis`` keyword to return an array with axis dimension(s)
+        removed. The default is ``'median'``.
+
+    stdfunc : {'std', 'mad_std'} or callable, optional
+        The statistic or callable function/object used to compute the
+        standard deviation about the center value. If using a callable
+        function/object and the ``axis`` keyword is used, then it must
+        be able to ignore NaNs (e.g., `numpy.nanstd`) and it must have
+        an ``axis`` keyword to return an array with axis dimension(s)
+        removed. The default is ``'std'``.
+
+    axis : None or int or tuple of int, optional
+        The axis or axes along which to sigma clip the data. If `None`,
+        then the flattened data will be used. ``axis`` is passed to the
+        ``cenfunc`` and ``stdfunc``. The default is `None`.
+
+    grow : float or `False`, optional
+        Radius within which to mask the neighbouring pixels of those
+        that fall outwith the clipping limits (only applied along
+        ``axis``, if specified). As an example, for a 2D image a value
+        of 1 will mask the nearest pixels in a cross pattern around each
+        deviant pixel, while 1.5 will also reject the nearest diagonal
+        neighbours and so on.
+
+    Notes
+    -----
+    The best performance will typically be obtained by setting
+    ``cenfunc`` and ``stdfunc`` to one of the built-in functions
+    specified as a string. If one of the options is set to a string
+    while the other has a custom callable, you may in some cases
+    see better performance if you have the `bottleneck`_ package
+    installed. To preserve accuracy, bottleneck is only used for float64
+    computations.
+
+    .. _bottleneck:  https://github.com/pydata/bottleneck
+
+    See Also
+    --------
+    sigma_clipped_stats, SigmaClip, sigma_clip
+    """
+
+    def __init__(
+        self,
+        data: ArrayLike,
+        mask: NDArray | None = None,
+        mask_value: float | None = None,
+        sigma: float | None = 3.0,
+        sigma_lower: float | None = None,
+        sigma_upper: float | None = None,
+        maxiters: int | None = 5,
+        cenfunc: Literal["median", "mean"] | Callable | None = "median",
+        stdfunc: Literal["std", "mad_std"] | Callable | None = "std",
+        axis: int | tuple[int, ...] | None = None,
+        grow: float | Literal[False] | None = False,
+    ) -> None:
```

Comment:
since this is a brand new class, we have the opportunity to make most arguments keyword-only from the get go, and I think we should. I would start at `mask` (so, only *data* is allowed positionally)

Reply:
done, thanks